src/functions/updateUploadStatus/framework/handler.ts

```diff
@@ -7,13 +7,14 @@ import { isNullOrBlank } from '../../../functions/postResult/framework/handler';
 import { updateUpload } from '../application/update-upload-service';
 import { error, warn, bootstrapLogging } from '@dvsa/mes-microservice-common/application/utils/logger';
 import { InconsistentUpdateError } from '../domain/InconsistentUpdateError';
+import { SubmissionOutcome } from '../domain/SubmissionOutcome';
 
 export async function handler(event: APIGatewayProxyEvent, fnCtx: Context): Promise<Response> {
 
   bootstrapLogging('update-upload-status', event);
   await bootstrapConfig();
 
-  let body: string;
+  let body: SubmissionOutcome;
```

Comment:
Are we sure what `body` will be always "Convertable" to this structure? 
We do have ```body = JSON.parse(event.body);``` at some point

Reply:
I would hope so as in the queryBuilder we accept this` body: any` and then this with no checks 
`
  uploadStatus = body.state;
  retryCount = body.retry_count;
  errorMessage = body.error_message;
  applicationReference = appReference;
  staffNumber = body.staff_number;
  uploadInterface = body.interface;`

On the getNextBatch lambda we have enjoi validation on the query string. I think in this case we need to push this type out into schemas/common microservice and use the same one across the 3 interfaces and do some validation after the parse. I think it's going to add too much of a delay for getting this one out and it's on my notes list of things I noticed doing this ticket